framework/src/org/checkerframework/common/util/count/TypeStatistics.java

```diff
@@ -0,0 +1,79 @@
+package org.checkerframework.common.util.count;
+
+import com.sun.source.tree.ArrayAccessTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.NewArrayTree;
+import com.sun.source.tree.ParameterizedTypeTree;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import org.checkerframework.framework.source.SourceChecker;
+import org.checkerframework.framework.source.SourceVisitor;
+import org.checkerframework.javacutil.AnnotationProvider;
+
+/**
+ * An annotation processor for listing the facts about types and array accesses locations of
```

Comment:
Document what this does -- what "the facts" are.
Generic types feel rather different than array accesses.  I would put each into its own class, or jam everything into AnnotationStatistics with different command-line arguments.  (Maybe this class is the kitchen-sink class for everything except annotations, though?  In that case, the "Type" in its name is a bit misleading.  Array access locations aren't really related to types.)